diff --git a/node_modules/slonik/dist/types.d.ts b/node_modules/slonik/dist/types.d.ts
index 5bf1d51..2a02b60 100644
--- a/node_modules/slonik/dist/types.d.ts
+++ b/node_modules/slonik/dist/types.d.ts
@@ -204,8 +204,13 @@ export declare type ValueExpressionType = SqlTokenType | PrimitiveValueExpressio
 export declare type NamedAssignmentType = {
     readonly [key: string]: ValueExpressionType;
 };
-export declare type SqlTaggedTemplateType<T = QueryResultRowType> = {
-    <U = T>(template: TemplateStringsArray, ...values: ValueExpressionType[]): TaggedTemplateLiteralInvocationType<U>;
+
+
+export type QueryParameters<T> = T extends {'@params': any[], } ? T['@params'] : ValueExpressionType[];
+export type QueryResult<T> = Omit<T, '@params'>;
+
+export type SqlTaggedTemplateType<T = QueryResultRowType> = {
+    <U = T>(template: TemplateStringsArray, ...values: QueryParameters<U>): TaggedTemplateLiteralInvocationType<QueryResult<U>>,
     array: (values: ReadonlyArray<PrimitiveValueExpressionType>, memberType: TypeNameIdentifierType | SqlTokenType) => ArraySqlTokenType;
     binary: (data: Buffer) => BinarySqlTokenType;
     identifier: (names: ReadonlyArray<string>) => IdentifierSqlTokenType;
